backend/modules/simulation/core/orchestrator.py

```
@@ -0,0 +1,699 @@
+        return self.scenes[self.state.current_scene_index].get('success_criteria', 'User completes interaction')
+    
+    def _get_turns_remaining(self) -> int:
+        """Calculate turns remaining for current scene"""
+        if not self.scenes or self.state.current_scene_index >= len(self.scenes):
+            return 0
+        
+        scene = self.scenes[self.state.current_scene_index]
+        # Use timeout_turns, require it to be set
+        timeout_turns = scene.get('timeout_turns')
+        if not timeout_turns:
+            raise ValueError("Scene must have timeout_turns configured")
```

Ruff (0.14.8)

Avoid specifying long messages outside the exception class

(TRY003)